### PR TITLE
Add block names to error message for block restriction

### LINKED ISSUE #18506: Have block/boundary name for block/boundary restrictable error
## Reason
When a sideset or a block is missing, the interface will return some id number that is beyond the existing ones. That is super unhelpful

## Design
Quick lookup for the id and get the name at the same index. Easy easy

## Impact
Save some time when debugging people s inputs

### PR DESCRIPTION
**Pull request by GiudGiud**
closes #18506

### CODE REVIEW

File: test/tests/restrictable/undefined_ids/tests [hunk 1]
```diff
@@ -2,7 +2,7 @@
   [block]
     type = 'RunException'
     input = 'undefined_block_kernel.i'
-    expect_err = "the following block ids do not exist on the mesh: 10"
+    expect_err = "the following block (ids) do not exist on the mesh: 10 (10)"
```
Comment (dschwen) [inline test/tests/restrictable/undefined_ids/tests:5]: I think it just adds confusion if the name is the ID and gets duplicated in the parenthesis

Comment (GiudGiud) [reply test/tests/restrictable/undefined_ids/tests:5]: umm so if the name is the id then I just output the id

**Comment by GiudGiud** [comment]
ok test failure is linked to use of boundary restriction and face variables. I ll see if I can debug this at some point.

**Comment by GiudGiud** [comment]
Ok in order to debug this PR's test failures, which seems unrelated to the Griffin test, I went to valgrind Griffin and with upstream/devel I am getting tons of error messages for that particular test. @YaqiWang @zachmprince could you please confirm

**Comment by GiudGiud** [comment]
I could use a map to remove those 3 vectors. I'd be ok with that change but this could also be merged as it.

**Review by lindsayad — APPROVED** [review]

**Pull request merged**